### PR TITLE
Add id field to vkb device

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: tools/libxl/libxl_vkb.c [hunk 1]
```diff
@@ -28,6 +28,18 @@ static int libxl__device_vkb_dm_needed(libxl_device_vkb *vkb, uint32_t domid)
     return 0;
 }
 
+static int libxl__set_xenstore_vkb(libxl__gc *gc, uint32_t domid,
+                                   libxl_device_vkb *vkb,
+                                   flexarray_t *back, flexarray_t *front,
+                                   flexarray_t *ro_front)
+{
+    if (vkb->id) {
```
Comment (andr2000) [inline tools/libxl/libxl_vkb.c]: no brackets around single expression?

File: tools/xl/xl_parse.c [hunk 2]
```diff
@@ -1110,7 +1110,10 @@ int parse_vkb_config(libxl_device_vkb *vkb, char *token)
             return -1;
         }
         vkb->backend_type = backend_type;
-    } else {
+    } else if (MATCH_OPTION("id", token, oparg)) {
```
Comment (andr2000) [inline tools/xl/xl_parse.c]: pls check coding style

File: tools/xl/xl_parse.c [hunk 3]
```diff
@@ -1151,7 +1154,7 @@ static void parse_vkb_list(const XLU_Config *config,
 
             if (vkb->backend_type == LIBXL_VKB_BACKEND_UNKNOWN) {
                 fprintf(stderr, "backend-type should be set in vkb spec\n");
-                rc = -1; goto out;
+                rc = ERROR_FAIL; goto out;
```
Comment (andr2000) [inline tools/xl/xl_parse.c]: new line?

File: tools/libxl/libxl_vkb.c (hunk 1 above)
Comment (al1img) [reply tools/libxl/libxl_vkb.c]: Every indented statement is braced, but blocks that contain just one
statement may have the braces omitted.

File: tools/xl/xl_parse.c (hunk 3 above)
Comment (al1img) [reply tools/xl/xl_parse.c]: Error code paths with an if statement and a goto or a return on the same
line are allowed.

**Pull request merged**